nav2_tasks/CMakeLists.txt

```diff
@@ -68,6 +68,5 @@ endif()
 
 ament_export_include_directories(include)
 ament_export_libraries(${library_name})
-ament_export_dependencies(${dependencies})
```

Comment:
@nuclearsandwich Thanks for pointing this out. We do need to export at least `tf2_geometry_msgs`. It is a dependency of `spin_action.hpp` and is required downstream when building `nav2_bt_navigator`. A better approach for now:

``` ament_export_dependencies(tf2_geometry_msgs) ```

And as you suggested, add it to the `package.xml`:

``` build_export_depend(tf2_geometry_msgs) ```

Reply:
I'll implement this now

Reply:
> why don't you just add `tf2_geometry_msgs` to the package xml in bt navigator rather than making it the tasks package's problem?

If this package is publishing messages provided by another package, such that any package which consumes this one will need those messages, I think exporting the dependency here is a valid thing to do.

Reply:
Ok, @orduno can you do an exhaustive search to see if there's any others we're missing?

Reply:
@nuclearsandwich should we more generally be removing these ament export dependencies? They're broadly used right now https://github.com/ros-planning/navigation2/search?q=ament_export_dependencies&unscoped_q=ament_export_dependencies

Reply:
>  should we more generally be removing these ament export dependencies? They're broadly used right now https://github.com/ros-planning/navigation2/search?q=ament_export_dependencies&unscoped_q=ament_export_dependencies


*Good catch!* Each of those should definitely be looked at, ideally by someone familiar with the package to determine whether exported packages are actually needed by downstream or not. Those that aren't can be removed and those that are can be audited to make sure there's a corresponding build_export_depend in the package.xml.

Reply:
Yeah. We kind of took a blunt hammer approach and exported everything in many cases, and apparently didn't do it right (need the build_export_depend). But I don't think we should remove it altogether.

Without the ament_export_dependencies line, somebody will be forced to manually figure out all the build failures they'll get whenever they include a header from one of our packages.

Reply:
I'm not sure we're saying to remove them altogether as much as audit them and make sure we've resolved this issue as made famous by `rosidl`